index.d.ts

```diff
@@ -0,0 +1,54 @@
+// Project: Immutability helper
+// TypeScript Version: 2.2
+
+export = update
+
+// declare function update<T>(data: T, query: update.Query<T>): T
+declare function update<T>(
+  data: ReadonlyArray<T>,
+  query: ArrayOperators<T>,
+): ReadonlyArray<T>
+
+declare function update<T>(data: T, query: Query<T>): object
```

Comment:
This should be the last overload so that the `ReadonlySet` and `ReadonlyMap` ones are tried first.